krabs/krabs/filtering/event_filter.hpp

```diff
@@ -123,9 +142,14 @@ namespace krabs {
             return;
         }
 
-        if (!predicate_(record)) {
-            return;
-        }
+		if(predicate_ != nullptr) //only if predicate exists... else trust the native filtering
```

Comment:
nit: here and elsewhere add a space after the `if` keyword: `if (foo == bar)`

Reply:
Done